run-branch-mutation-tests.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+# Executes Pitest to determine the mutation test coverage of changed code. By
+# default code that is changed relative to the default branch is exercised,
+# though another change set may be specified; see the Arcmutate Git integration
+# documentation for details [1]. The results are found in each Maven module's
+# `target/pit-reports` directory.
+#
+# [1] https://docs.arcmutate.com/docs/git-integration.html
+
+set -e -u -o pipefail
+
+if [ "${#}" -gt 1 ]; then
+  echo "Usage: ${0} [DiffSpec]"
```

Comment:
I think we want to have something like `opt:` in here right? In case of optional arguments we internally usually do:
`<opt: diff-spec>`.

Reply:
I see we don't do it for the other, but now we worked a lot more with bash so we made some improvements here and there.

Reply:
Hmm, [square brackets represent optionality](https://en.wikipedia.org/wiki/Usage_message#Pattern); this alternative is less idiomatic IMHO.